jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -151,47 +155,40 @@ ecma_builtin_regexp_dispatch_helper (const ecma_value_t *arguments_list_p, /**<
   }
 #endif /* ENABLED (JERRY_ES2015) */
 
+  ecma_value_t ret_value;
   ecma_object_t *new_target_obj_p = ecma_op_regexp_alloc (new_target_p);
 
-  if (JERRY_UNLIKELY (new_target_obj_p == NULL))
+  if (JERRY_LIKELY (new_target_obj_p != NULL))
   {
 #if ENABLED (JERRY_ES2015)
-    if (free_arguments)
+    if (create_regexp_from_bc)
     {
-      ecma_free_value (pattern_value);
-      ecma_free_value (flags_value);
+      ret_value = ecma_op_create_regexp_from_bytecode (new_target_obj_p, bc_p);
```

Comment:
It seems this can never throw. If that's the case we could move the `ECMA_IS_VALUE_ERROR ` check into the else case, and leave an assert here.

Reply:
True, will do the update

Reply:
@dbatyai I've updated the PR.